contracts/Template.sol

```diff
@@ -0,0 +1,104 @@
+/*
+ * SPDX-License-Identitifer:    GPL-3.0-or-later
+ *
+ * This file requires contract dependencies which are licensed as
+ * GPL-3.0-or-later, forcing it to also be licensed as such.
+ *
+ * This is the only file in your project that requires this license and
+ * you are free to choose a different license for the rest of the project.
+ */
+
+pragma solidity 0.4.24;
+
+import "@aragon/os/contracts/factory/DAOFactory.sol";
+import "@aragon/os/contracts/apm/Repo.sol";
+import "@aragon/os/contracts/lib/ens/ENS.sol";
+import "@aragon/os/contracts/lib/ens/PublicResolver.sol";
+import "@aragon/os/contracts/apm/APMNamehash.sol";
+
+import "@aragon/apps-voting/contracts/Voting.sol";
+import "@aragon/apps-token-manager/contracts/TokenManager.sol";
+import "@aragon/apps-shared-minime/contracts/MiniMeToken.sol";
+
+import "./CounterApp.sol";
+
+contract KitBase is APMNamehash {
+    ENS public ens;
+    DAOFactory public fac;
+
+    event DeployInstance(address dao);
+    event InstalledApp(address appProxy, bytes32 appId);
+
+    function KitBase(DAOFactory _fac, ENS _ens) {
+        ens = _ens;
+
+        // If no factory is passed, get it from on-chain bare-kit
+        if (address(_fac) == address(0)) {
+            bytes32 bareKit = apmNamehash("bare-kit");
+            fac = KitBase(latestVersionAppBase(bareKit)).fac();
+        } else {
+            fac = _fac;
+        }
+    }
+
+    function latestVersionAppBase(bytes32 appId) public view returns (address base) {
+        Repo repo = Repo(PublicResolver(ens.resolver(appId)).addr(appId));
+        (,base,) = repo.getLatest();
+
+        return base;
+    }
+}
+
+contract Kit is KitBase {
+    MiniMeTokenFactory tokenFactory;
+
+    uint64 constant PCT = 10 ** 16;
+    address constant ANY_ENTITY = address(-1);
+
+    function Kit(ENS ens) KitBase(DAOFactory(0), ens) {
+        tokenFactory = new MiniMeTokenFactory();
+    }
+
+    function newInstance() {
+        Kernel dao = fac.newDAO(this);
+        ACL acl = ACL(dao.acl());
+        acl.createPermission(this, dao, dao.APP_MANAGER_ROLE(), this);
+
+        address root = msg.sender;
+        bytes32 appId = apmNamehash("app");
+        bytes32 votingAppId = apmNamehash("voting");
+        bytes32 tokenManagerAppId = apmNamehash("token-manager");
+
+        CounterApp app = CounterApp(dao.newAppInstance(appId, latestVersionAppBase(appId)));
```

Comment:
Let's add a **BIG** comment here saying this is the place they should change their app name.